### ISSUE TITLE
Enable JavaScript and forms in response preview

### ISSUE THREAD

**Issue by uicosp** [a85/POSTMan-Chrome-Extension#2646]
1. Postman Version: 4.9.3
 2. App (Chrome app or Mac app): Mac app
 3. OS details: 10.12.2
 4. Is the Interceptor on and enabled in the app: No
 5. Did you encounter this recently, or has this bug always been there: N/A
 6. Expected behaviour: N/A
 7. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app): N/A
 8. Screenshots (if applicable) N/A

For some reason, I need run javascript and submit forms in *Preview*. I know postman doesn't support it for security ([CSP](https://content-security-policy.com/)). So I inspected postman's source code via DevTools, and finally I found 

```
131078                 { className: 'response-body-iframe-viewer' },
131079                 _react2.default.createElement('iframe', {
131080                   className: 'response-body-viewer-preview',
131081                   sandbox: '' // https://developer.mozilla.org/en/docs/Web/HTML/Element/iframe#attr-sandbox
131082                   , src: this.props.src
131083                 })
```

in `/Applicatoins/Postman.app/Contents/Resources/app.js`

Then I modify line 131081 to 
`sandbox: 'allow-forms allow-scripts'`

relaunch postman, it works!

I think it's may not safe when using postman under `sandbox: 'allow-forms allow-scripts'` all the time. So maybe we could add a button to postman's preferences settings, then we can enable/disable it as needed.

**Comment by shamasis** [comment]
This one is tricky. Not everybody will understand the implications like you do. If we do implement this, we've got to make it clear of the security aspects that it breaks down.

Maybe the feature could be a one time thing. That is, in preview tab, a button saying re-render in less secure mode. Thinking aloud.

What's your thought?

**Comment by uicosp** [comment]
@shamasis It's a good idea, I agree with you.

**Comment by StijnVansevenant** [comment]
This is also an issue on the Windows Desktop version which we would like to see resolved. Thanks.

To elaborate on the issue. We want to use postman for API calls of an onprem Dynamics CRM server, which is exposed to the cloud using Azure Application Proxy. This is a service by Azure that allows your "ADFS & Windows Based Authentication onprem servers" to be consumed through a public endpoint that uses OAuth claimes based authentication with bearer tokens.

Because of how the authentication works on the Azure Application Proxy and how the authentication works on our onprem Dynamics CRM API's, we are getting the "Script is disabled" error. Here's more detail info on that particular situation: https://social.technet.microsoft.com/wiki/contents/articles/1425.ad-fs-2-0-script-is-disabled-click-submit-to-continue.aspx

Basically we require Postman to accept javascript. With a simple, permanent, configuration setting.

**Comment by blueelvis** [comment]
@shamasis - Sam, any chance this is on your radar? This would be really helpful.

**Comment by jhmaster2000** [comment]
> 1. Postman Version: 4.9.3
> 2. App (Chrome app or Mac app): Mac app
> 3. OS details: 10.12.2
> 4. Is the Interceptor on and enabled in the app: No
> 5. Did you encounter this recently, or has this bug always been there: N/A
> 6. Expected behaviour: N/A
> 7. Console logs (http://blog.getpostman.com/2014/01/27/enabling-chrome-developer-tools-inside-postman/ for the Chrome App, View->Toggle Dev Tools for the Mac app): N/A
> 8. Screenshots (if applicable) N/A
> 
> For some reason, I need run javascript and submit forms in _Preview_. I know postman doesn't support it for security ([CSP](https://content-security-policy.com/)). So I inspected postman's source code via DevTools, and finally I found
> 
> ```
> 131078                 { className: 'response-body-iframe-viewer' },
> 131079                 _react2.default.createElement('iframe', {
> 131080                   className: 'response-body-viewer-preview',
> 131081                   sandbox: '' // https://developer.mozilla.org/en/docs/Web/HTML/Element/iframe#attr-sandbox
> 131082                   , src: this.props.src
> 131083                 })
> ```
> 
> in `/Applicatoins/Postman.app/Contents/Resources/app.js`
> 
> Then I modify line 131081 to
> `sandbox: 'allow-forms allow-scripts'`
> 
> relaunch postman, it works!
> 
> I think it's may not safe when using postman under `sandbox: 'allow-forms allow-scripts'` all the time. So maybe we could add a button to postman's preferences settings, then we can enable/disable it as needed.

Hello, Im trying to reproduce your steps to enable JS on Preview, but I cant find the files you refer to? ("/Applicatoins/Postman.app/Contents/Resources/app.js") can you give me a more detailed walkthrough on it? Thanks.

**Comment by ngondard** [comment]
In Postman 7.1.0, I found almost the same line:
` sandbox: sandboxRestrictions // https://developer.mozilla.org/en/docs/Web/HTML/Element/iframe#attr-sandbox`
This line belongs to the following file:
`app/ressources/app/js/vendor-shared.js`
Not sure about how to use it, though.

**Comment by nikandlv** [comment]
This can make it work https://github.com/nikandlv/postman-patcher

**Comment by stefanJ-hub** [comment]
Hi,

sorry for cross-posting. Initial comment went into: https://github.com/postmanlabs/postman-app-support/issues/5803

I am on Windows / v7.21.1

"We use Postman a lot, especially the integrated feature to get an Oauth Token via the Auth Code Flow.
Now our IDS changed and the login form is submitted via javascript.
This - similar to the topics mentioned above - does not work anymore now. Not able to complete the auth code flow with postman.

The deprecated Postman Chromeapp seems to do still do the job, but comes with its own drawbacks.. and: is deprecated.

The patch from @nikandlv only works for linux/mac versions.
I understand security concerns, but couldn't this be made toggleable (as above mentioned patch is doing) - with disabled just being the default?

Would appreciate the ability to be able to have javascript enabled in the Oauth Login pages rendered by Postman.

If I am just missing a setting, or if you know a workaround / patch for windows - please give me a hint."

**Comment by ghost** [comment]
I also need this possibilty that @stefanJ-hub mentioned. would be great if this problem gets resolved. Thank you!

**Comment by AndyRiedel** [comment]
I'm having the same issue as @stefanJ-hub but on Windows 10

**Comment by a-gehlot** [comment]
As a new user who ran into this problem when testing an HTML form, and didn't realize what it was for hours, could someone explain to me what the security issue is in layman's terms? Why does Chrome allow the HTML submit button to work properly if it's a security issue? Thanks.

**Comment by nikandlv** [comment]
This issue has been going for more than 3 years. i made a patch for it last year and worked well but the updates broke it.

Its a truly an important feature that we need and postman team is not listening.
All we demand is a toggleable setting for this mode for savvy people 
Till then me and my team moved to **[insomnia](https://insomnia.rest/)** and its been a breeze 

So simply they don't care what we need, they care about what they want to implement and what gets more attraction.

1. i will update the patcher later
2. i will write my rant in a blog post soon
3. Even if you finally do implement it, it has passed more then **3** years for it

this https://github.com/postmanlabs/postman-app-support/projects/32#card-34481266 is not acceptable, this necessary feature takes less then a day to implement. even writing the patcher took just couple of hours and yet the issue is being tracked for 3 years

@kamalaknn @shamasis @numaanashraf @saswatds

**Comment by timsamart** [comment]
We are currently implementing Okta into our pipeline. This is basically the biggest player in User-Authentication in the world. Without this feature we are not able to reliably test Authentication. I would suspect that many of the big players who also use Okta either:

1. use an old version of Postman or
2. are not using Postman

It would be cool to implement this. Thanks. 

Also anyone who has the same problem should also post here in order to display his interest and blow up the thread! It would be more productive to follow a **cando** attitude then a **blame** one.

**Comment by arlemi** [comment]
Hey @timsamart, appreciate the support for the feature request, though we would not recommend for everyone to post here, unless there's useful information around the use case that can be added. Posting "+1"s and "me too"s only sends unnecessary notifications to everyone and doesn't add any value for the engineers when trying to implement the feature. 🙂 

To show your support, add a 👍 on the first post.

**Comment by timsamart** [comment]
> Hey @timsamart, appreciate the support for the feature request, though we would not recommend for everyone to post here, unless there's useful information around the use case that can be added. Posting "+1"s and "me too"s only sends unnecessary notifications to everyone and doesn't add any value for the engineers when trying to implement the feature. 🙂 
> 
> To show your support, add a 👍 on the first post. 

Hey @arlemi yes it would be better to just add the thumbs up 🙏

**Comment by Shelby86** [comment]
So you made your app totally useless and still won't fix it

**Comment by amrit-hop** [comment]
We have a SAML use-case that we'd like to put forward as a good reason to at least give the user the option to enable Javascript in the Preview window.

Basically, we are using an IdP-initiated flow which means, we GET a SAMLResponse token and submit it to an ACS URI through an auto-executing SAML-standard form, that looks like this:

```
<html>
<head>
    <meta http-equiv='X-UA-Compatible' content='IE=edge' />
    <base target='_self'/>
    <title>Working...</title>
</head>
<body>
<form action='https://*******.com/auth/saml/borrower' method='post'>
        <input type='hidden' name='SAMLResponse' value='PHNhbW...[truncated for brevity]'/>
    <noscript>
        <p><strong>Note:</strong> Since your browser does not support JavaScript,
        you must press the Continue button once to proceed.</p>
        <input type='submit' value='Continue'/>
    </noscript>
</form>
<script>window.addEventListener('load', function(){document.forms[0].submit();});</script>
</body>
</html>
```

This form won't work in the Preview window because Javascript is disabled.  If it were enabled, and it was allowed to execute this form, it would be submitted to the action url automatically upon rendering it.  Then, that would complete the IdP-initiated flow, and as long as cookies were coming through and being saved, this request to GET a SAMLResponse should result in either our IdP's login page or the HTML for the landing page of the protected resource that we just SSO'd into by executing the above form if a cookie was persisted already that had the SSO creds.

Insomnia has you beat, Postman.  They allow this feature.  But their application is buggy and doesn't properly get rid of the cookies it captures.  So, I thought for sure Postman would do it right.  Well, I was wrong. :)

You should allow this feature as an option for people to use at their own risk.  You can't be liable for someone executing javascript in your Preview window, can you?

**Comment by gertkix** [comment]
Also have several saml and IDP setups that cannot be tested via postman. Any update on this one?

**Comment by 200-0K** [comment]
Any updates?

**Comment by Fighteros** [comment]
Any updates?

**Comment by kiangti** [comment]
Any updates?

**Comment by Arkemlar** [comment]
Any updates?

**Comment by dko1905** [comment]
Any updates?

**Comment by hatefi-arman** [comment]
Dear Postman team, 
The use case we have is regarding integrating other applications for an onboarding scenario that a combination of API calls and page visits are needed. The result of one step should be presented to the other step and a Postman collection is really helpful for keeping all in one place and transferring result from one step to another. Unfortunately the disabled Javascript execution in Postman response window is blocking to have such collection. 
I appreciated if this feature could be added. Thanks.
